### PR TITLE
RCIAM 852 , RCIAM 851 and RCIAM-883

### PR DESCRIPTION
**Pull request by laskasn**
This PR fulfills the objectives of  the following tickets:

- [RCIAM-852](https://jira.argo.grnet.gr/browse/RCIAM-852)
- [RCIAM-851](https://jira.argo.grnet.gr/browse/RCIAM-851)
- [RCIAM-883](https://jira.argo.grnet.gr/browse/RCIAM-883)

### CODE REVIEW

File: roles/keycloak/templates/15.0.2/standalone-ha.xml.j2 [hunk 1]
```diff
@@ -123,7 +124,7 @@
                 <pattern-formatter pattern="%d{yyyy-MM-dd HH:mm:ss,SSS} %-5p [%c] (%t) %s%e%n"/>
             </formatter>
             <formatter name="COLOR-PATTERN">
-                <pattern-formatter pattern="%K{level}%d{yyyy-MM-dd HH:mm:ss,SSS} %-5p [%c] (%t) %s%e%n"/>
+                <pattern-formatter pattern="%K{level}%d{HH:mm:ss,SSS} %-5p [%c] (%t) %s%e%n"/>
```
Comment (NicolasLiampotis) [inline roles/keycloak/templates/15.0.2/standalone-ha.xml.j2:127]: It's better to leave the yyyy-MM-dd information in each log line (log files can be sent to remote syslog, concatenated processed etc)

File: roles/keycloak/tasks/main.yml [hunk 2]
```diff
@@ -207,11 +229,12 @@
       src: templates/keycloak.service.j2
       dest: /etc/systemd/system/keycloak.service
     become: yes
+    notify:
+      - Reload systemd daemon
```
Comment (NicolasLiampotis) [inline roles/keycloak/tasks/main.yml:233]: Reload keycloak

File: roles/keycloak/handlers/main.yml [hunk 3]
```diff
@@ -1,19 +1,26 @@
...
   become: yes
 
 - name: Restart keycloak
-  service: 
+  systemd:
     name: keycloak
     state: restarted
   become: yes
+
+- name: Reload systemd daemon
```
Comment (NicolasLiampotis) [inline roles/keycloak/handlers/main.yml:21]: Reload keycloak

File: roles/keycloak/defaults/main.yml [hunk 4]
```diff
@@ -37,6 +37,7 @@ keycloak_bind_port: "8080"
 
 keycloak_java_opts: "-Xms1024m -Xmx2048m -XX:MaxPermSize=768m"
 
+keycloak_logs_max_days: 540  # that's ~18 months
```
Comment (NicolasLiampotis) [inline roles/keycloak/defaults/main.yml:40]: 548 (18 months = 547.863298 days)

File: roles/keycloak/tasks/main.yml [hunk 5]
```diff
@@ -185,6 +185,28 @@
...
+        src: templates/logcleaner.sh.j2
+        dest: "{{ keycloak_home }}/logcleaner.sh"
+    - name: "Create the cronjob"
+      ansible.builtin.cron:
+        name: keycloak logcleaner
+        day: "*"
+        hour: "0"
+        minute: "0"
+        user: "{{ keycloak_service_user }}"
+        job: "find {{ keycloak_home }}/standalone/log -type f -name 'server.log.[0-9][0-9][0-9][0-9]-[0-9][0-9]-[0-9][0-9]*' -exec sh {{ keycloak_home }}/logcleaner.sh {} {{ keycloak_logs_max_days }} \\;"
```
Comment (NicolasLiampotis) [inline roles/keycloak/tasks/main.yml:205]: The script should take keycloak log dir as parameter and use find

File: roles/keycloak/templates/logcleaner.sh.j2 [hunk 6]
```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+
+## call this script with the following command
+## find <FOLDER_OF_LOGS> -type f -name 'server.log.[0-9][0-9][0-9][0-9]-[0-9][0-9]-[0-9][0-9]*' -exec sh logcleaner.sh {} <DAYS-TO-KEEP-LOGS> \;
+## so, we assume that $1 holds the filename fullpath and $2 holds the maximum days from today to keep
```
Comment (NicolasLiampotis) [inline roles/keycloak/templates/logcleaner.sh.j2:5]: The script should call find on the log directory

**Review by NicolasLiampotis — CHANGES_REQUESTED** [review]

File: roles/keycloak/templates/15.0.2/standalone-ha.xml.j2 (hunk 1 above)
Comment (laskasn) [reply roles/keycloak/templates/15.0.2/standalone-ha.xml.j2:127]: you 're right

**Comment by laskasn** [comment]
@nliam This needs to be reviewed again

**Review by NicolasLiampotis — APPROVED** [review]
LGTM

**Pull request merged**